### ISSUE TITLE
Verifying docker images on default profile is not meaningful

### ISSUE THREAD

**Issue by imesh** [abhishek0198/wso2dockerfiles-test-framework#11]
The default profile [1] of WSO2 docker images do not have any platform specific configurations. They are very much similar to the vanilla product distributions. Ideally those would work always without any problem because each product release goes through a QA process. 

Therefore the test framework would either need to do distributed deployments of WSO2 products on standalone docker or on a container cluster manager such as K8S or DC/OS. Thanks!

[1] https://github.com/wso2/puppet-modules/blob/master/hieradata/dev/wso2/wso2as/5.3.0/default/default.yaml

**Comment by abhishek0198** [comment]
@imesh 
Makes sense. IMHO, the purpose of testing with default profile is to test the Dockerfiles codebase, to make sure that we are able to create docker images, run them and ensure that the product launch using Docker container is successful. I agree, that we need to add support to test distributed deployments, however, having a default platform as the starting point for the test framework was essential so that we can start small and add capabilities over iterations.

**Comment by chamilad** [comment]
I think the `default` profile is part of the code base that should be covered by a test. The main reason for this is that we have Puppet configs, Dockerfile configs etc that handle the `default` profile.

**Comment by imesh** [comment]
Yes, I agree that it should be the starting point. However without verifying product profiles with distributed deployments, testing framework would not provide much value to the test automation process. I consider that as the key feature of this test framework. Thanks!

**Comment by chamilad** [comment]
+1 The core of the test framework should be to cover distributed profiles.
